server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/WebSocketRoutes.scala

```diff
@@ -45,8 +50,23 @@ object WebSocketRoutes {
   val LOGGER: Logger = LoggerFactory.getLogger(classOf[WebSocketRoutes])
 }
 
+case class ClientContext(outbound: Sinks.Many[WebSocketOutboundMessage], pushRegistration: AtomicReference[Registration], session: MailboxSession) {
+
+  def withRegistration(registration: Registration): Unit = Option(pushRegistration.getAndSet(registration))
+    .foreach(oldRegistration => SMono.fromCallable(() => oldRegistration.unregister())
+      .subscribeOn(Schedulers.elastic())
+      .subscribe())
+
+  def clean(): Unit =
+    Option(pushRegistration.getAndSet(null))
+      .foreach(registration => SMono.fromCallable(() => registration.unregister())
+        .subscribeOn(Schedulers.elastic())
+        .subscribe())
```

Comment:
Can we factorize stuff with a `private def  withRegistration(registration: Option[Registration]): Unit` ?